src/codegen/generators/expression_generator.rs

```diff
@@ -200,10 +200,15 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
         }) = self.annotations.get(expression)
         {
             if !self.index.get_type_information_or_void(resulting_type).is_aggregate() {
-                // constant propagation
-                return self.generate_constant_expression(qualified_name, expression);
+                // Constant Propagation
+                if let Ok(expr) = self.generate_constant_expression(qualified_name, expression) {
+                    // We return here if constant propagation worked, and if not fall-back to generating the expression
+                    // further down which may or may not work but loads the values if it does
+                    return Ok(expr);
+                }
```

Comment:
Let's warn at this stage just to express that we are not ignoring the returned error.